### PR TITLE
Make third party stubtest output more succinct

### PR DESCRIPTION
**Pull request by srittau**
stubtest_third_party.py will now only print a single line when the test
succeeds for a certain distribution, unless the "-v" option is given.
When the test fails or "-v" is given, will still print the full output.

The success status is now colored to make spotting failures easier.

stdout/stderr is now used consistently: The distribution name and
success status is always printed to stdout, all other output goes to
stderr. Running the script with "2>/dev/null" will only show the success
status of the test, one per line.

### CODE REVIEW

**Comment by srittau** [comment]
Example output:

```
DateTimeRange... success
Deprecated... fail

/tmp/tmpl4towucr/bin/pip install Deprecated==1.2.* mypy==0.950
MYPYPATH=/home/srittau/Projekte/typeshed/stubs/Deprecated /tmp/tmpl4towucr/bin/python -m mypy.stubtest --ignore-missing-stub --custom-typeshed-dir /home/srittau/Projekte/typeshed deprecated --allowlist /home/srittau/Projekte/typeshed/stubs/Deprecated/@tests/stubtest_allowlist.txt

error: deprecated.classic.ClassicAdapter.__init__ is inconsistent, runtime does not have argument "abc"
Stub: at line 13
def (self: deprecated.classic.ClassicAdapter, reason: builtins.str =, version: builtins.str =, action: Union[Union[Literal['default'], Literal['error'], Literal['ignore'], Literal['always'], Literal['module'], Literal['once']], None] =, category: Type[builtins.Warning] =, abc: builtins.int =)
Runtime: at line 86 in file /tmp/tmpl4towucr/lib/python3.9/site-packages/deprecated/classic.py
def (self, reason='', version='', action=None, category=<class 'DeprecationWarning'>)


Ran with the following environment:
Deprecated==1.2.13
mypy==0.950
mypy-extensions==0.4.3
tomli==2.0.1
typing-extensions==4.2.0
wrapt==1.14.1

To fix "unused allowlist" errors, remove the corresponding entries from /home/srittau/Projekte/typeshed/stubs/Deprecated/@tests/stubtest_allowlist.txt

Flask-Cors... success
Flask-SQLAlchemy... success
JACK-Client... success
```

**Review by AlexWaygood — COMMENTED** [review]
Looks good. I think we should add types-termcolor to our `requirements-tests.txt` file, though, or `mypy_test.py` will fail when run locally.

**Comment by srittau** [comment]
> Looks good. I think we should add types-termcolor to our `requirements-tests.txt` file, though, or `mypy_test.py` will fail when run locally.

Maybe we should just add `# type: ignore[import]`?

**Comment by AlexWaygood** [comment]
> > Looks good. I think we should add types-termcolor to our `requirements-tests.txt` file, though, or `mypy_test.py` will fail when run locally.
> 
> Maybe we should just add `# type: ignore[import]`?

Since we're only importing one function, we could also just do:

```python
if TYPE_CHECKING:
    def colored(s: str, _: str) -> str: ...
else:
  try:
      from termcolor import colored
  except ImportError:
  
      def colored(s: str, _: str) -> str:
          return s

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-      - run: pip install $(grep tomli== requirements-tests.txt) $(grep mypy== requirements-tests.txt)
+      - run: pip install $(grep tomli== requirements-tests.txt) $(grep mypy== requirements-tests.txt) types-termcolor
```
[the remark this answers is not in the document]
Comment (srittau) [reply .github/workflows/tests.yml]: Good, as I didn't like that install anyway.

**Review by AlexWaygood — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]
Looks good, but let's give @hauntsaninja a chance to comment too.

**Review by hauntsaninja — APPROVED** [review]
Looks good!

**Pull request merged**